datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -944,6 +949,53 @@ impl ExecutionPlan for HashJoinExec {
             try_embed_projection(projection, self)
         }
     }
+
+    fn gather_filters_for_pushdown(
+        &self,
+        phase: FilterPushdownPhase,
+        parent_filters: Vec<Arc<dyn PhysicalExpr>>,
+        _config: &ConfigOptions,
+    ) -> Result<FilterDescription> {
+        if self.join_type != JoinType::Inner {
```

Comment:
I recommend we  file a ticket and then add a reference to that ticket in the code a ticket reference so someone in the future who stumbles on this code can locate the relevant context

Reply:
https://github.com/apache/datafusion/issues/16973